plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java

```diff
@@ -36,16 +36,44 @@
 public class CodeTemplates {
 
   /**
-   * Load the named template into the supplied Eclipse project.
-   *  
+   * Creates files for a sample App Engine Standard project in the supplied Eclipse project.
+   *
    * @param project the Eclipse project to be filled with templated code
    * @param config replacement values
    * @param monitor progress monitor
-   * @param name directory from which to load template
    * @return the most important file created that should be opened in an editor
    */
-  public static IFile materialize(IProject project, AppEngineProjectConfig config,
+  public static IFile materializeAppEngineStandardFiles(IProject project, AppEngineProjectConfig config,
       IProgressMonitor monitor) throws CoreException {
+    return materialize(project, config, true /* isStandardProject */, monitor);
+  }
+
+  /**
+   * Creates files for a sample App Engine Flexible project in the supplied Eclipse project.
+   *
+   * @param project the Eclipse project to be filled with templated code
+   * @param config replacement values
+   * @param monitor progress monitor
+   * @return the most important file created that should be opened in an editor
+   */
+  public static IFile materializeAppEngineFlexFiles(IProject project, AppEngineProjectConfig config,
+      IProgressMonitor monitor) throws CoreException {
+    return materialize(project, config, false /* isStandardProject */, monitor);
+  }
+
+  /**
+   * Creates files for a sample App Engine project in the supplied Eclipse project.
+   *
+   * @param project the Eclipse project to be filled with templated code
+   * @param config replacement values
+   * @param isStandardProject true if project should be configured to have the App Engine Standard
+   *   configuration files and false if project should have the App Engine Flexible configuration
+   *   files.
+   * @param monitor progress monitor
+   * @return the most important file created that should be opened in an editor
+   */
+  private static IFile materialize(IProject project, AppEngineProjectConfig config,
```

Comment:
Yes, and I am suggesting this should be two methods too. Consider using methods as the way to choose between two code paths rather than conditionals.